src/pages/EventPage/EventPage.tsx

```diff
@@ -20,12 +20,54 @@ type AddAvailabilityModalProps = {
     heatMapData: HeatMapData;
     show: boolean;
     onHide: React.MouseEventHandler<HTMLButtonElement> | undefined;
+    eventAvailability: EventDataAvailability;
+    eventId: string;
+    uid: string;
 };
 
+function appendUserAvailabilityToGroup(
+    eventAvailability: EventDataAvailability,
+    userAvailability: number[][],
+    uid: string
+): EventDataAvailability {
+    const yTimes = Object.keys(eventAvailability).sort();
+    const xDays = Object.keys(
+        eventAvailability[Object.keys(eventAvailability)[0]]
+    ).sort();
+
+    for (let i = 0; i < userAvailability.length; i += 1) {
+        for (let j = 0; j < userAvailability[0].length; j += 1) {
+            if (userAvailability[i][j] === 1) {
```

Comment:
The logic for appending user and event availability is a bit complex here. I'd suggest doing a refactor with implementation and breaking logic down if possible or commenting the logic. Either works but I suspect if the logic was documented a refactor for readability and efficiency would become apparent.